### PR TITLE
support different models of ENCODE_rE2G

### PR DESCRIPTION
**Pull request by atancoder**
We can infer the E2G model to use based on the biosamples config.

Add 2 types of models: DHS + intact HiC, DHS + megamap
along with their feature table, feature table config, and thresholds

Incorporated many of the script changes @mayasheth suggested in https://github.com/EngreitzLab/ENCODE_rE2G/pull/9
- made some minor changes and code cleanups from that PR

## Test Plan

Ran the biosamples_config chr22 end to end

### CODE REVIEW

File: workflow/rules/utils.smk [hunk 1]
```diff
@@ -14,3 +18,42 @@ def make_paths_absolute(obj, base_path):
 			return new_file
 	return obj
 
+def _get_biosample_model_dir(biosample):
```
Comment (atancoder) [inline workflow/rules/utils.smk:21]: @mayasheth here's where we map the biosample row to the model. We'll update this function as more models get supported

File: README.md [hunk 2]
```diff
@@ -18,3 +18,16 @@ snakemake -j1 --use-conda
...
 
 Output will show up in the `results/` directory
+
+
+## Supported Models
+
+We have pre-trained ENCODE-rE2G on certain model types. You can find them in the `models` directory.
+Each model must have the following:
+1. model pickle file
+2. feature table config
```
Comment (atancoder) [inline README.md:28]: @mayasheth seems like we also need a feature table config file for each model?

Comment (atancoder) [reply README.md:28]: I'm going to assume the feature table config that we use for dhs_intact_hic and dhs_megamap are the same?

Comment (mayasheth) [reply README.md:28]: Yes: feature table/config determines what features are required for a specific model and what they're called, which is needed to create final feature table. also ensures the names of the features in the trained model align with the feature tables.

Yes, feature tables for dhs_intact_hic and dhs_megamap are the same since we are pulling from the same ABC contact column and we decided to use the same set of features.

File: config/config.yaml [hunk 3]
```diff
@@ -9,23 +9,19 @@ results_dir: "results/"
...
+crispr_dataset: "reference/EPCrisprBenchmark_ensemble_data_GRCh38.tsv.gz"
 
 # ENCODE E2G
-feature_table:
-  ATAC: "/oak/stanford/groups/engreitz/Users/sheth/ENCODE-E2G_v2/data/feature_table/model_scATAC_IGVF.tsv"
-  DHS: "/oak/stanford/groups/engreitz/Users/atan5133/ENCODE-E2G/data/feature_table/model_ablation_encode_e2g.tsv"
-models_dir:
-  ATAC: "/oak/stanford/groups/engreitz/Users/sheth/ENCODE-E2G_v2/models_scATAC_IGVF_filt/K562_cellranger_1/"
-  DHS: "/oak/stanford/groups/engreitz/Users/atan5133/ENCODE-E2G/models"
+MEGAMAP_HIC_FILE: https://s3.us-central-1.wasabisys.com/aiden-encode-hic-mirror/bifocals_iter2/tissues.hic
```
Comment (mayasheth) [inline config/config.yaml:20]: why is this needed here? shouldn't this be part of the ABC module?

File: models/dhs_intact_hic/feature_table_config.tsv
Comment (mayasheth) [inline models/dhs_intact_hic/feature_table_config.tsv:1]: i don't think this is needed, it's the same as feature_table? the "nice_name" column is just for plotting and stuff in the future, but they are identical otherwise

Comment (mayasheth) [reply models/dhs_intact_hic/feature_table_config.tsv:1]: sorry, let me clarify: feature_table_config is not needed: there are a few features that aren't actually needed by the model so they don't need to / should not be included in the final feature table, as that could be confusing. for example, we don't need 3DContact_squared or TargetGenePromoterActivityQuantile.

File: models/dhs_megamap/feature_table_config.tsv
Comment (mayasheth) [inline models/dhs_megamap/feature_table_config.tsv:1]: again- not needed.

File: workflow/scripts/gen_new_features.py [hunk 4]
```diff
@@ -37,17 +38,20 @@ def main(enhancer_list, abc_predictions, ref_gene_tss, chr_sizes, results_dir):
 
     determine_num_candidate_enh_gene(pred_df, results_dir)
     determine_num_tss_enh_gene(pred_df, ref_gene_tss, results_dir, intermediate_dir)
-    generate_num_sum_enhancers(abc_predictions, enhancer_list, chr_sizes, results_dir, intermediate_dir)
+    generate_num_sum_enhancers(
```
Comment (mayasheth) [inline workflow/scripts/gen_new_features.py:41]: for this function/feature, do we want to make the distance range we compute over a parameter? we had previously compared within 5kb and within 10kb and chosen 5kb (i think?) but i believe we still compute both features-- for now we could just compute the 5kb versions since this is a computationally expensive step?

File: config/config.yaml (hunk 3 above)
Comment (atancoder) [reply config/config.yaml:20]: We use this to determine to use the megamap model. We compare the hic_file in the biosamples config. See `_get_biosample_model_dir` fn in utils.smk

File: workflow/scripts/gen_new_features.py (hunk 4 above)
Comment (atancoder) [reply workflow/scripts/gen_new_features.py:41]: That makes sense. That's probably better to do as a separate PR

Comment (atancoder) [reply workflow/scripts/gen_new_features.py:41]: actually i'll just remove the 10kb in this PR since I'm already re-running the pipeline for testing anyways

**Comment by atancoder** [comment]
Addressed comments. Support final features module for scE2G

**Review by mayasheth — APPROVED** [review]
thanks for implementing these changes!
